spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/aggregation/AggregationOperationContext.java

```diff
@@ -114,4 +118,9 @@ default Fields getFields(Class<?> type) {
 	default AggregationOperationContext continueOnMissingFieldReference() {
 		return this;
 	}
+
+	@Override
+	default CodecRegistry getCodecRegistry() {
+		return MongoClientSettings.getDefaultCodecRegistry();
```

Comment:
In practice, when (if ever) will this default implementation be used?  The underlying driver never uses this method, because it doesn't take into account important user configuration (most notably, the UUID representation).

Reply:
Since it’s public API that users may have implemented, defaulting here eases the pain when upgrading to the new version. Internal implementations pick up the driver registry.
Are custom codecs heavily used?

Reply:
The main thing is that even the default registry is changed in the MongoClient constructor in order to apply the uuidRepresentation to the registry.  The default registry will throw when trying to encode a UUID value, because the default UuidRepresentation is UNSPECIFIED

Reply:
Just a note: `MongoClientSettings.getDefaultCodecRegistry()` is also used explicitly in the default implementation of the `MongoConverter.getCodecRegistry` method.

Reply:
true, same reason for those upgrading. As long as the user does not provide a custom implementation of the converter the codec registry (with all the user settings regarding UUID codec,...) will be used.